src/Access/AccessControl.cpp

```diff
@@ -653,6 +654,23 @@ bool AccessControl::isPlaintextPasswordAllowed() const
     return allow_plaintext_password;
 }
 
+void AccessControl::setDefaultPasswordTypeFromConfig(const String & type_)
+{
+    if (type_ == "plaintext")
+        default_password_type = AuthenticationType::PLAINTEXT_PASSWORD;
+    else if (type_ == "double_sha1")
+        default_password_type = AuthenticationType::DOUBLE_SHA1_PASSWORD;
+    else if (type_ == "sha256")
+        default_password_type = AuthenticationType::SHA256_PASSWORD;
+    else
+        throw Exception(ErrorCodes::UNKNOWN_ELEMENT_IN_CONFIG, "Unknown password type in 'default_password_type' in config");
```

Comment:
Maybe we can use `AuthenticationTypeInfo` somehow, `AccessControl.cpp` is too general cpp file for such things.